lib/kube/proxy/self_subject_reviews.go

```diff
@@ -271,15 +287,21 @@ func (m *kubernetesResourceMatcher) Match(role types.Role, condition types.RoleC
 	kind := m.resource.Kind
 	name := m.resource.Name
 	namespace := m.resource.Namespace
-	apiGroup := m.resource.APIGroup
-	if apiGroup == "" {
-		apiGroup = "core"
+	// If the resource is global, clear the namespace.
+	// NOTE: kubectl will yield a warning for this case, but we still need to process the request.
+	if m.isClusterWideResource {
+		namespace = ""
+	}
+
+	// If we are dealing with a namespace resource, consider it cluster wide even though it is not.
+	if m.resource.Kind == "namespaces" {
+		m.isClusterWideResource = true
 	}
 
 	for _, resource := range resources {
 		isResourceTheSameKind := kind == resource.Kind || resource.Kind == types.Wildcard
 
-		namespaceScopeMatch := resource.Kind == "namespaces" && slices.Contains(types.KubernetesNamespacedResourceKinds, kind+"."+apiGroup)
+		namespaceScopeMatch := resource.Kind == "namespaces" && !m.isClusterWideResource
```

Comment:
To fix v7 -> v8 behavior with the namespace, I am unsure if we should move `namespace` -> to `namespaces` behavior. Should we allow both where we allow `v7+namespace` meaning the namespace + resources within that namespace while `v8+namespaces` means only namespace resource this fixing it for the future

Reply:
v8 doesn't change the namespace behavior, that may come in v9 based on the RBAC RFD.

Although it would be interesting to keep `namespace` with the existing v7 behavior and have `namespaces` be a only for the namespace resource indeed. Is that what you are suggesting?

Reply:
Exactly so we don't need the role v9

Reply:
Done in a separate PR https://github.com/gravitational/teleport/pull/54938